tests/integration/security/jwt_test.go

```diff
@@ -409,61 +358,79 @@ func TestIngressRequestAuthentication(t *testing.T) {
 				policy := tmpl.EvaluateAllOrFail(t, namespaceTmpl, file.AsStringOrFail(t, filename))
 				t.Config().ApplyYAMLOrFail(t, ns.Name(), policy...)
 			}
-
 			applyPolicy("testdata/requestauthn/global-jwt.yaml.tmpl", rootNS{})
-			applyPolicy("testdata/requestauthn/ingress.yaml.tmpl", ns)
-
-			bSet := apps.B.Match(echo.Namespace(ns.Name()))
 
 			callCount := 1
 			if t.Clusters().IsMulticluster() {
 				// so we can validate all clusters are hit
 				callCount = util.CallsPerCluster * len(t.Clusters())
 			}
 
-			for _, cluster := range t.Clusters() {
-				t.NewSubTest(fmt.Sprintf("In %s", cluster.StableName())).Run(func(t framework.TestContext) {
-					a := apps.A.Match(echo.InCluster(cluster).And(echo.Namespace(apps.Namespace1.Name())))
-					// These test cases verify in-mesh traffic doesn't need tokens.
-					testCases := []authn.TestCase{
-						{
-							Name: "in-mesh-with-expired-token",
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   bSet[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Headers: map[string][]string{
-										authHeaderKey: {"Bearer " + jwt.TokenExpired},
-									},
-									Count: callCount,
-								},
-								DestClusters: bSet.Clusters(),
+			t.NewSubTest("in-mesh-authn").Run(func(t framework.TestContext) {
+				// These test cases verify in-mesh traffic doesn't need tokens.
+				testCases := []authn.TestCase{
+					{
+						Name: "in-mesh-with-expired-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + jwt.TokenExpired},
 							},
-							ExpectResponseCode: response.StatusUnauthorized,
+							Count: callCount,
 						},
-						{
-							Name: "in-mesh-without-token",
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   bSet[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Count:    callCount,
-								},
-								DestClusters: bSet.Clusters(),
-							},
-							ExpectResponseCode: response.StatusCodeOK,
+						ExpectResponseCode: response.StatusUnauthorized,
+					},
+					{
+						Name: "in-mesh-without-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Count:    callCount,
 						},
-					}
-					for _, c := range testCases {
-						t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
-							retry.UntilSuccessOrFail(t, c.CheckAuthn,
-								retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second))
+						ExpectResponseCode: response.StatusCodeOK,
+					},
+				}
+				for _, c := range testCases {
+					echotest.New(t, apps.All).
+						SetupForPair(func(t framework.TestContext, src, dst echo.Instances) error {
+							policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
```

Comment:
can you create the policy once and reuse it here and below (line 424)?

Reply:
I don't think this could be merged into one setup. The policy setup here is triggered for each subtest, `src` and `dst` vary in every subtest (`a` or `vm`). The flow is `src`->`dst`
line 424 is choosing `b` and the flow is `ingr->ingress gateway->b`

Reply:
I mean before adding the `dst` to the template, the only variable that makes a different in the policy is the `Namespace`, which seems to be able to be created early and then applied later in each test.

Reply:
Sorry about causing the confusion. I have templatized the "testdata/requestauthn/ingress.yaml.tmpl" by replacing hardcoded `b` with `{{ .dst }}`, and later reverted the change by mistake. We have to apply the policy here and below because we set different destination to the policy for those subtests.